app/src/main/java/com/pokemonshowdown/app/BattleFieldActivity.java

```diff
@@ -359,6 +359,11 @@ public void onClick(DialogInterface dialog, int which) {
                 final String userName = intent.getExtras().getString(MyApplication.EXTRA_LOGIN_SUCCESSFUL);
                 Toast.makeText(this, String.format(getResources().getString(R.string.login_successful), userName), Toast.LENGTH_SHORT).show();
                 break;
+
+            case MyApplication.EXTRA_REPLAY_DATA:
+                final String replayData = intent.getExtras().getString(MyApplication.EXTRA_REPLAY_DATA);
+                new ExportReplayTask(this).execute(replayData);
```

Comment:
Why do you need to route it through `MyApplication`? Can't you just execute it in BattleFragment?